samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMsgMetadata.java

```diff
@@ -51,36 +50,37 @@
 
 
   /**
-   * The timestamp of when the events actually happened
-   * set by and copied from the event source
+   * The timestamp (EpochMilli) when the event happened
+   * set by and copied from the event's source
    * TODO: copy eventTime through from source to RelMessage
    */
   @JsonProperty("eventTime")
   private long eventTime;
 
   /**
    * the timestamp of when Samza App received the event
-   * TODO: set arrivalTime during conversion from IME to SamzaMessage
+   * TODO: set arrivalSystemNanoTime during conversion from IME to SamzaMessage
    */
-  @JsonProperty("arrivalTime")
```

Comment:
Does this JSON content get persisted anywhere across deployments? If so, then an upgraded system will not be able to read old data, since you renamed this JSON property.

Reply:
I checked some Jackson tutorials and checked some code and it seems that Jackson's ObjectMapper uses org.codehaus.jackson.map, which doesn't seem to be persisted.. in the coming up update, arrivalTime won't be renamed, but scanTime is, however, I am using same JsonProperty value for backward compatibility